### PR TITLE
Feature/document search

### PR DESCRIPTION
**Pull request by afdezl**

### CODE REVIEW

**Comment by afdezl** [comment]
NEEDS: 
- Documentation
- Tests

File: app/airview_api/search.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    def query(self, search_term: str):
+        body = {
+            "query": {
+                "multi_match": {  # This will returned a ranked search.
+                    "query": search_term
+                }
+            }
+        }
+
+        search_result = self.client.search(
```
Comment (afdezl) [inline app/airview_api/search.py:76]: Will need to handle exceptions (i.e. Timeouts)

File: app/airview_api/controllers/search.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+@blp.route("/")
+class Search(MethodView):
+    @blp.response(200, SearchResultSchema(many=True))
+    @blp.role(Roles.CONTENT_READER)
+    def get(self):
+        """
+        Searches matching terms in a configured search backend.
+        """
+        results = []
+        q = request.args.get("q", None)
```
Comment (afdezl) [inline app/airview_api/controllers/search.py:28]: Needs opt params to handle search and result behaviour (i.e.: number of returned results, wildcards..)

Comment (mcuckson) [reply app/airview_api/controllers/search.py:28]: Have a look to see if the flask-smorest schema stuff will help, there's an exaple - https://github.com/AirWalk-Digital/airview-api/blob/main/app/airview_api/controllers/technical_controls.py#L34. It allows for deserialization of query params into the schema and handles all the param validation, etc.

File: app/airview_api/search.py (hunk 1 above)
Comment (afdezl) [reply app/airview_api/search.py]: Let it raise a 500 for now

**Review by mcuckson — APPROVED** [review]

File: app/airview_api/search.py (hunk 1 above)
Comment (mcuckson) [reply app/airview_api/search.py]: 500 is fine, not sure what else we can do about timeouts anyway and the frontend will put up a friendly 'unexpected error, try again' for it anyway, which is all the user can really do.

**Pull request merged**